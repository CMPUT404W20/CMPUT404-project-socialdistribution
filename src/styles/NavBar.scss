$accent-color: #0275b1;

$badge-bg-color: #ffb75e;

$placeholder-text-color: #bababa;

$icon-text-color: #181818;

$border-color: #d9d9d9;

.selected {
  color: $accent-color !important;
}

.icons > a,
.user > a,
.log-out > a {
  text-decoration: none;
  color: $icon-text-color;

  &:hover,
  &:focus,
  &:active {
    color: $accent-color;
  }

  & > p {
    font-style: normal;
    font-weight: 400;
    font-size: 8pt;
    margin-top: 0;
    margin-bottom: 0;
  }
}
// .form-control:focus {
//   outline: 0;
//   border: 0;
// }
.searchBar {
  .form-control {
    
  }
  .form-control:focus {
    border-color: unset;
    box-shadow: unset;
  }
  .btn-outline-secondary {
    background-color: white;
    // outline: 0;
    border: 1px red solid;

    &:hover {
      background-color: red;
    }
  }
  width: 100%;
}

.searchButton {
  color: $accent-color;
  vertical-align: middle;
}

.header {
  background-color: white;
  width: 100%;
  .navbar {
      .logo {
          font-weight: 700;
          font-size: 18px;
      }
      padding: 0 !important;

      .nav-link {
          padding-right: 0 !important;
          padding-left: 35px !important;
          font-size: 12px;
          text-align: center;
          font-weight: 500;
          color: black;
          line-height: 10px;

          &:hover {
              color: black;
          }
      }
      
      .navbar-toggler {
          border: none !important;
          padding: 0 !important;
          font-size: 14px !important;
          outline: 0 !important;
      }

      @media screen and (max-width: 576px) {
          .nav-link {
              padding-left: 0px !important;
          }
      }
  }
}



.nav {
  // background-color: white;
  // position: fixed;
  // justify-content: center;
  // top: 0;
  // width: 100%;
}

.navRow {
  display: flex;
  height: 40px;
  width: 100%;
  align-content: center;
  justify-content: center;
  flex-direction: row;
  padding-top: 0.07rem;
}

.left-menu {
  justify-content: center;
  display: flex;
  flex-direction: row;
  width: 50%;
}

.logo-container {
  display: inline-flex;
  align-items: center;
}

.search-input-container,
.right-side-menu,
.user,
.log-out {
  display: flex;
  border-left: 1px solid $border-color;
  align-items: center;
  height: 40px;
  padding-left: 2%;
}

.search-input-container {
  width: 50%;
}

.search-input {
  border: none;
  font-size: 10pt;
  margin-left: 2%;
  width: 80%;
}

::-webkit-input-placeholder,
::-moz-placeholder {
  color: $placeholder-text-color;
  font-style: normal;
  font-weight: normal;
}

.right-side-menu {
  padding-left: 2%;
  width: 50%;
}

.icons {
  display: flex;
  align-items: center;
  padding-left: 1%;
}

.log-out {
  padding-left: 2%;
}

.icons > a {
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  margin-right: 2rem;
}

.logout-div {
  display: flex;
  flex-shrink: 0;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}

.user {
  margin-left: 1rem;
  margin-right: 10%;
  padding-left: 3%;
}

.icon-wrapper {
  display: flex;
  justify-content: center;
}

.badge-wrapper {
  background: $badge-bg-color;
  width: 10pt;
  height: 10pt;
  border-radius: 50%;
  display: flex;
  justify-content: center;
  margin-left: -0.6rem;
}

.badge {
  font-weight: 700;
  color: white !important;
  font-size: 6pt;
  align-self: center;
}