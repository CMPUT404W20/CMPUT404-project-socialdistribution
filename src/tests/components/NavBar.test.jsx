import React from "react";
import { shallow } from "enzyme";
import NavBar from "../../components/NavBar";

describe("Navigation bar Component", () => {
  it("should render correctly", () => {
    const component = shallow(<NavBar />);
    expect(component).toMatchSnapshot();
  });

  it("should not update class name when not being selected", () => {
    const component = shallow(<NavBar />);
    expect(component.find(".selected")).toHaveLength(0);
  });

  it("should update class name on selected", () => {
    const component = shallow(<NavBar selected="Home" />);
    expect(component.find(".selected")).toHaveLength(1);
  });
});
