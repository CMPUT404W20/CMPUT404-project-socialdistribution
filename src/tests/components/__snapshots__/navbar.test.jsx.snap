// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar Component should render correctly 1`] = `
<div
  className="nav"
>
  <div
    className="navRow"
  >
    <div
      className="left-menu"
    >
      <div
        className="logo-container"
      >
        <img
          alt="app logo"
          src="logo.svg"
          width="70%"
        />
      </div>
      <div
        className="search-input-container"
      >
        <SearchRoundedIcon
          style={
            Object {
              "color": "#0275B1",
            }
          }
        />
        <input
          className="search-input"
          onKeyDown={[Function]}
          placeholder="Search"
        />
      </div>
    </div>
    <div
      className="right-side-menu"
    >
      <div
        className="icons"
      >
        <a
          className="Home"
          href=" "
        >
          <HomeOutlinedIcon />
          <p>
            HOME
          </p>
        </a>
        <a
          className="Friends"
          href=" "
        >
          <PeopleAltOutlinedIcon />
          <p>
            FRIENDS
          </p>
        </a>
        <a
          className="Notices"
          href=" "
        >
          <div
            className="icon-wrapper"
          >
            <NotificationsNoneOutlinedIcon />
            <div
              className="badge-wrapper"
            >
              <span
                className="badge"
              >
                1
              </span>
            </div>
          </div>
          <p>
            NOTICES
          </p>
        </a>
      </div>
      <div
        className="user"
      >
        <a
          className=""
          href=" "
        >
          <p>
            Username
          </p>
        </a>
      </div>
      <div
        className="log-out"
      >
        <a
          className="logout-div"
          href=" "
        >
          <ExitToAppOutlinedIcon />
          <p>
            LOG OUT
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
